File: Programmers/Level5/hanoi(cjpt)/Hanoi.java
import java.util.Arrays;

class Hanoi {
	public int[][] helpHanoi(int n, int from, int via, int to) {
        if(n==1)
        {
        	int[][] answer = {{from,to}};
    		return answer;
        }

        int[][] answer = new int[(int)Math.pow(2,n)-1][2];
        int[][] temp = helpHanoi(n-1,from,to,via);
        for(int i=0; i<temp.length; ++i)
        {
        	answer[i] = temp[i];
        }
        int[] temp2 = {from,to};
        answer[temp.length] = temp2;
        temp = helpHanoi(n-1,via,from,to);
        for(int i=0; i<temp.length; ++i)
        {
        	answer[i+temp.length+1] = temp[i];
        }
		return answer;
	}

	public int[][] hanoi(int n)
	{
		return helpHanoi(n,1,2,3);
	}

	public static void main(String[] args) {
		Hanoi h = new Hanoi();
		System.out.println(Arrays.toString(h.hanoi(4)));
	}
}

File: Programmers/Level5/hanoi(cjpt)/hanoi.py
def hanoi(n):
    if n==1:
        return [[1,3]]
    temp1 = []
    temp2 = []
    for mov in hanoi(n-1):
        temp1.append(list(map(lambda x: ((2*x)%3+1)%3+1,mov)))
        temp2.append(list(map(lambda x: ((x+1)*2)%3+1,mov)))
    temp1.append([1,3])
    temp1.extend(temp2)
    return temp1  # 2차원 배열을 반환해 주어야 합니다.


# 아래는 테스트로 출력해 보기 위한 코드입니다.
print(hanoi(3))